ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -14033,12 +14033,26 @@ ABoundActivity * HqlCppTranslator::doBuildActivityDedup(BuildCtx & ctx, IHqlExpr
 
     if (!useHash)
     {
-        if (info.compareAllRows)
-            doBuildBoolFunction(instance->classctx, "compareAll", info.compareAllRows);
-        if (!info.keepLeft)
-            doBuildBoolFunction(instance->classctx, "keepLeft", info.keepLeft);
+        // Only necessary to create getFlags() if different from base class getFlags() - i.e. different from default
+        if ( !(!info.compareAllRows && !info.keepBest && info.keepLeft) )
+        {
+            StringBuffer flags;
+            if (info.compareAllRows)
+                flags.append("|DFcompareall");
+            if (info.keepLeft)
+                flags.append("|DFkeepleft");
+            if (info.keepBest)
+                flags.append("|DFkeepbest");
+            if (flags.length()==0)
+                flags.append("|0");
```

Comment:
This is for the situation where compareAllRows=false, keepBest=false and keepLeft=false. i.e. all the flags are OFF.  Base class getflags() returns DFkeepleft ... so if keepleft is false (and the other flags are also false), the function has to return 0.